spec/requests/groups_controller_spec.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+describe GroupsController do
+  let(:user)  { create(:user) }
+  let(:group) { create(:group) }
+  before      { login_as(user) }
+
+  describe '#index' do
+    it 'shows active groups list' do
+      get '/groups'
+
+      expect(response).to have_http_status(:ok)
+      expect(response).to render_template(:index)
+    end
+  end
+
+  describe '#all' do
+    it 'shows all groups list' do
+      get '/groups/all'
+
+      expect(response).to have_http_status(:ok)
+      expect(response).to render_template(:index)
+    end
+  end
+
+  describe '#show' do
+    it 'shows a group' do
+      get "/groups/#{group.id}"
+
+      expect(response).to have_http_status(:ok)
+    end
+  end
+
+  describe '#edit' do
+    context 'when the user is not the leader' do
+      it 'returns forbidden' do
+        get "/groups/#{group.id}/edit"
+
+        expect(response).to have_http_status(:forbidden)
+      end
+    end
+
+    context 'when the user is the group leader' do
+      let(:leader) { group.leader.user }
+      before       { login_as(leader) }
+      it 'renders the page' do
+        get "/groups/#{group.id}/edit"
+
+        expect(response).to have_http_status(:ok)
+        expect(response).to render_template(:edit)
+      end
+    end
+  end
+
+  describe '#update' do
+    context 'when the user is not the leader' do
+      it 'returns forbidden' do
+        patch "/groups/#{group.id}"
+
+        expect(response).to have_http_status(:forbidden)
+      end
+    end
+
+    context 'when the user is the group leader' do
+      let(:leader) { group.leader.user }
+      before       { login_as(leader) }
+      it 'updates the group' do
+        patch "/groups/#{group.id}", group: { description: 'babot hámoznak' }
+
+        expect(response).to redirect_to(group_path(group))
+        expect(group.reload.description).to eq 'babot hámoznak'
```

Comment:
```ruby
        group.reload
        expect(group.name).to eq('New name')
        expect(group.description).to eq('New description')
        expect(group.webpage).to eq('www.example.org')
```
and so on.

Also please keep the usage/not usage of parenthesis, but do not mix it.